components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,130 @@
+    pub const LOWER_V1: ResourceKey = resource_key!(uniset, "Lower", 1);
+    pub const MATH_V1: ResourceKey = resource_key!(uniset, "Math", 1);
+    pub const NCHAR_V1: ResourceKey = resource_key!(uniset, "NChar", 1);
+    pub const NFCINERT_V1: ResourceKey = resource_key!(uniset, "nfcinert", 1);
+    pub const NFDINERT_V1: ResourceKey = resource_key!(uniset, "nfdinert", 1);
+    pub const NFKCINERT_V1: ResourceKey = resource_key!(uniset, "nfkcinert", 1);
+    pub const NFKDINERT_V1: ResourceKey = resource_key!(uniset, "nfkdinert", 1);
+    pub const PAT_SYN_V1: ResourceKey = resource_key!(uniset, "Pat_Syn", 1);
+    pub const PAT_WS_V1: ResourceKey = resource_key!(uniset, "Pat_WS", 1);
+    pub const PCM_V1: ResourceKey = resource_key!(uniset, "PCM", 1);
+    pub const PRINT_V1: ResourceKey = resource_key!(uniset, "print", 1);
+    pub const QMARK_V1: ResourceKey = resource_key!(uniset, "QMark", 1);
+    pub const RADICAL_V1: ResourceKey = resource_key!(uniset, "Radical", 1);
+    pub const RI_V1: ResourceKey = resource_key!(uniset, "RI", 1);
+    pub const SD_V1: ResourceKey = resource_key!(uniset, "SD", 1);
+    pub const SEGSTART_V1: ResourceKey = resource_key!(uniset, "segstart", 1);
+    pub const SENSITIVE_V1: ResourceKey = resource_key!(uniset, "Sensitive", 1);
+    pub const STERM_V1: ResourceKey = resource_key!(uniset, "STerm", 1);
+    pub const TERM_V1: ResourceKey = resource_key!(uniset, "Term", 1);
+    pub const UIDEO_V1: ResourceKey = resource_key!(uniset, "UIdeo", 1);
+    pub const UPPER_V1: ResourceKey = resource_key!(uniset, "Upper", 1);
+    pub const VS_V1: ResourceKey = resource_key!(uniset, "VS", 1);
+    pub const WSPACE_V1: ResourceKey = resource_key!(uniset, "WSpace", 1);
+    pub const XDIGIT_V1: ResourceKey = resource_key!(uniset, "xdigit", 1);
+    pub const XIDC_V1: ResourceKey = resource_key!(uniset, "XIDC", 1);
+    pub const XIDS_V1: ResourceKey = resource_key!(uniset, "XIDS", 1);
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct UnicodeProperties<'s> {
+    pub props: Vec<UnicodeProperty<'s>>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct UnicodeProperty<'s> {
+    pub name: &'s str,
+    pub inv_list: Vec<u32>,
+}
+
+impl Default for UnicodeProperty<'static> {
+    fn default() -> UnicodeProperty<'static> {
+        UnicodeProperty {
+            name: "",
+            inv_list: vec![],
+        }
+    }
+}
+
+impl<'s> UnicodeProperty<'s> {
+    /// Default empty nameless property
+
+    pub fn from_uniset(set: &UnicodeSet, name: &'s str) -> UnicodeProperty<'s> {
+        let inv_list = set.get_inversion_list();
+        UnicodeProperty { name, inv_list }
+    }
+}
+
+impl<'s> TryInto<UnicodeSet> for UnicodeProperty<'s> {
+    type Error = crate::error::Error;
+    fn try_into(self) -> Result<UnicodeSet, Self::Error> {
+        UnicodeSet::from_inversion_list(self.inv_list)
+            .map_err(crate::error::Error::UnisetConversion)
+    }
+}
+
+#[test]
+fn test_uniset_to_inv_list() {
```

Comment:
Why is this test in this file instead of in the uniset crate?

Reply:
Done. Moved to uniset crate.